packages/@ourworldindata/explorer/src/Explorer.tsx

```diff
@@ -686,128 +666,111 @@ export class Explorer
-        //   1. find all variables that the transformed columns depend on and add them to
-        //      the config's dimensions array
-        //   2. download data and metadata of the variables
-        //   3. append the transformed columns to the Grapher table (note that this includes
-        //      intermediate columns that are defined for multi-step transforms but are not
-        //      referred to in any Grapher row)
-
-        // all slugs specified by the author in the explorer config
-        const uniqueSlugsInGrapherRow = _.uniq(
-            [...ySlugs.split(" "), xSlug, colorSlug, sizeSlug].filter(
-                _.identity
-            )
-        ) as string[]
+                    for (let index = 0; index < variableIds.length; index++) {
+                        const variableId = variableIds[index]
 
-        // find all variables that the transformed columns depend on and add them to the dimensions array
-        if (uniqueSlugsInGrapherRow.length) {
-            const baseVariableIds = _.uniq(
-                uniqueSlugsInGrapherRow.flatMap((slug) =>
-                    this.getBaseVariableIdsForColumnWithTransform(slug)
-                )
-            )
-                .map((id) => parseInt(id, 10))
-                .filter((id) => !isNaN(id))
-            baseVariableIds.forEach((variableId) => {
-                const hasDimension = dimensions.some(
-                    (d) => d.variableId === variableId
-                )
-                if (!hasDimension) {
-                    dimensions.push({
-                        variableId: variableId,
-                        property: DimensionProperty.table, // no specific dimension
-                    })
-                }
-            })
-        }
+                        const hasDimension = dimensions.some(
+                            (d) => d.variableId === variableId
+                        )
 
+                        // Add the dimension to the _beginning_ of the array
+                        // to ensure that the slug-based dimensions appear
+                        // before the variable id-based dimensions,
+                        // maintaining consistency with the ySlugs ordering
+                        // (see below)
+                        if (!hasDimension) {
+                            if (index === 0) {
+                                // If the transform depends on a single variable
+                                // (e.g. duplicate transform), then add that
+                                // variable as proper chart dimension. If the
+                                // transform depends on multiple variables, then
+                                // the first variable id is treated as the main
+                                // variable for the dimension.
+                                dimensions.unshift({
+                                    variableId: variableId,
+                                    property: dimension.property,
```

Comment:
This is an important change. Previously, all dimensions needed for a transformed columns were added as `table` property

Reply:
What is the purpose here?
I understand that having it as `table` will mean it doesn't show up if we just render the chart based on the `chart_config` alone; but if we have it like this, and have a column defined as `multiplyBy [varId] 2`, then it will show up wrong when rendered based on the `chart_config`, no?
Or is the purpose simply to get more correct metadata to show up?